lib/spack/spack/cmd/add.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import argparse
+
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.environment as ev
+
+
+description = 'add a spec to an environment'
+section = "environment"
+level = "long"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        'specs', nargs=argparse.REMAINDER, help="specs of packages to add")
+
+
+def add(parser, args):
+    env = ev.get_env(args, 'add')
+
+    for spec in spack.cmd.parse_specs(args.specs):
+        if not env.add(spec):
+            tty.msg("Package {0} was already added to {1}"
+                    .format(spec.name, env.name))
+        else:
+            tty.msg('Adding %s to environment %s' % (spec, env.name))
+    env.write()
```

Comment:
(for a later PR) ideally this would be wrapped in a context manager that automatically handled writing things out if an error didn't occur

Reply:
I agree with this.  If you want to do that refactor lemme know.  I think that maybe `with <environment> as env:` should handle that.